### PR TITLE
feat: [UIE-8128] - IAM RBAC add new account api with

### PR DESCRIPTION
**Pull request by aaleksee-akamai**
## Description 📝
IAM RBAC - new account api with

## Changes  🔄
- Added new api for Account
- No visible changes

## Target release date 🗓️
Nov 12, 2024 (dev)

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/api-v4/src/resources/types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export type AccountResource = {
+  resource_type: string;
```
Comment (bnussman-akamai) [inline packages/api-v4/src/resources/types.ts:2]: Can this be more specific?

For example:
```suggestion
  resource_type: 'account' | 'firewall';
```

File: packages/api-v4/src/resources/resources.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * getAccountResources
+ *
+ * Return all resources for account.
+ *
+ */
+export const getAccountResources = () => {
+  return Request<AccountResource>(
+    setURL(`${API_ROOT}/resources`),
```
Comment (bnussman-akamai) [inline packages/api-v4/src/resources/resources.ts:13]: Will IAM be in "Beta"? If so, we should use `API_BETA_ROOT`. If not, this is good!

**Review by bnussman-akamai — APPROVED** [review]
Looks good! 

There are a few things typed `string` or `string[]` that might make sense to make them an enum/string literal if possible so that our types are more strict. For example, grants are strictly typed. I'm not sure if it makes sense to try to do the same with IAM but it could be nice

https://github.com/linode/manager/blob/0da37095129c175e18351377f80f1086f596d70c/packages/api-v4/src/account/types.ts#L175-L214

File: packages/api-v4/src/resources/types.ts (hunk 1 above)
Comment (aaleksee-akamai) [reply packages/api-v4/src/resources/types.ts]: sure, I've updated it

File: packages/api-v4/src/iam/types.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export interface AccountPermissions {
+  account_access: Access[];
+  resource_access: Access[];
+}
+
+interface Access {
+  resource_type: string;
```
[the remark this answers is not in the document]
Comment (aaleksee-akamai) [reply packages/api-v4/src/iam/types.ts]: sure, thanks, i've updated that

File: packages/api-v4/src/iam/types.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  roles: Roles[];
+}
+
+export interface Roles {
+  name: string;
+  description: string;
+  permissions?: string[];
```
[the remark this answers is not in the document]
Comment (aaleksee-akamai) [reply packages/api-v4/src/iam/types.ts]: sure, done

File: packages/api-v4/.changeset/pr-11181-added-1730207508835.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/api-v4": Added
+---
+
+new account api ([#11181](https://github.com/linode/manager/pull/11181))
```
[the remark this answers is not in the document]
Comment (aaleksee-akamai) [reply packages/api-v4/.changeset/pr-11181-added-1730207508835.md]: updated to that comment, thanks

**Review by bnussman-akamai — APPROVED** [review]

**Review by cpathipa — COMMENTED** [review]
@aaleksee-akamai Overall, LGTM! Merging this [PR](https://github.com/linode/manager/pull/11146) caused some conflicts, it should be good to merge once the conflicts are resolved.

**Comment by aaleksee-akamai** [comment]
@cpathipa , I've resolved conflicts, thanks!

**Review by cpathipa — APPROVED** [review]

**Pull request merged**